### PR TITLE
[12.0] pos_loyalty

### PR DESCRIPTION
**Pull request by eLBati**
From https://github.com/OCA/pos/pull/414

### CODE REVIEW

**Comment by eLBati** [comment]
@geluboros could you give me an email address, recognized by github, to be set as author of a0758f09ab1f1f471dfb4ccc2e45179e726f19cc ?
Thanks

**Comment by eLBati** [comment]
@geluboros nevermind, found it

**Comment by eLBati** [comment]
The module seems to work.

Test are failing because the page `/pos/web` is blocked with `Connecting to the IoT Box`.

Does anybody know what module/configuration causes this? (@legalsylvain ?)
Tests of `pos_loyalty` are run without errors.

File: pos_loyalty/tests/test_pos_loyalty.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import odoo.tests
+
+
+class TestPOSLoyalty(odoo.tests.HttpCase):
+    def test_pos_loyalty(self):
+        env = self.env(user=self.env.ref('base.user_admin'))
+
+        main_pos_config = env.ref('point_of_sale.pos_config_main')
```
Comment (flotho) [inline pos_loyalty/tests/test_pos_loyalty.py:12]: Hi @elBati, thanks for your PR.
Maybe it's possible to unset iotbox option on this config
This should prevent the POS to look for the iotbox

**Review by flotho — CHANGES_REQUESTED** [review]

File: pos_loyalty/tests/test_pos_loyalty.py (hunk 1 above)
Comment (eLBati) [reply pos_loyalty/tests/test_pos_loyalty.py:12]: Tried with

```
diff --git a/pos_loyalty/tests/test_pos_loyalty.py b/pos_loyalty/tests/test_pos_loyalty.py
index 4901bec3..ce4671df 100644
--- a/pos_loyalty/tests/test_pos_loyalty.py
+++ b/pos_loyalty/tests/test_pos_loyalty.py
@@ -10,6 +10,15 @@ class TestPOSLoyalty(odoo.tests.HttpCase):
         env = self.env(user=self.env.ref('base.user_admin'))
 
         main_pos_config = env.ref('point_of_sale.pos_config_main')
+
+        # Disabile "use_proxy", see point_of_sale/static/src/js/models.js
+        main_pos_config.iface_payment_terminal = False
+        main_pos_config.iface_electronic_scale = False
+        main_pos_config.iface_print_via_proxy = False
+        main_pos_config.iface_scan_via_proxy = False
+        main_pos_config.iface_cashdrawer = False
+        main_pos_config.iface_customer_facing_display = False
+
         target_product = env.ref('point_of_sale.letter_tray')
         free_product = env.ref('point_of_sale.desk_organizer')
         customer = env.ref('base.res_partner_2')
```

but still getting

```
            <div class="loader">
                <div class="loader-feedback">
                    <h1 class="message">Connecting to the IoT Box</h1>
                    <div class="progressbar">
                        <div class="progress" width="50%" style="width: 23%;"></div>
                    </div>
                    <div class="button skip">
                        Skip
                    </div>
                </div>
            </div>
```

Comment (eLBati) [reply pos_loyalty/tests/test_pos_loyalty.py:12]: Also found `pos_customer_display` module, trying with 6ee9bcb2ce4304fe1f972fce7d39d50d207d935d

Comment (legalsylvain) [reply pos_loyalty/tests/test_pos_loyalty.py:12]: I think that merging https://github.com/OCA/pos/pull/763 could fix your tests.

**Comment by hitrosol** [comment]
any plan to make the travis green ?
it seems not an easy one...  😢

**Comment by wpichler** [comment]
@legalsylvain I don't get the point here - the first problem now is resolved, but not it seems to have problems on adding the product to the basket. As it seems (from the html output) - there is no product at all. Locally the tests are running all ok...

I can see that the demo products got loaded - so no idea what happens here....

**Comment by dsolanki-initos** [comment]
@eLBati can you please fix failed checks?

**Comment by dsolanki-initos** [comment]
@eLBati can you please check failed checks?

**Comment by eLBati** [comment]
Hi @dsolanki-initos . I am sorry, I am not going to work on this soon

**Pull request closed without merging**